frontend/components/SetupPage/SetupWelcome.tsx

```diff
@@ -181,30 +186,60 @@ export const SetupWelcomeLogin = () => {
     [updateBalances],
   );
 
+  const isServiceCreatedForAgent = useMemo(() => {
+    if (!isServicesFetched) return false;
+    if (!services) return false;
+    if (!selectedService) return false;
+    if (!selectedAgentConfig) return false;
+
+    return services.some(
+      (service) =>
+        service.home_chain ===
+        asMiddlewareChain(selectedAgentConfig.evmHomeChainId),
+    );
+  }, [isServicesFetched, services, selectedService, selectedAgentConfig]);
+
   useEffect(() => {
     if (!canNavigate) return;
-
-    // Navigate only when wallets and balances are loaded
     if (!isBalanceLoaded) return;
+    if (!isWalletsFetched) return;
 
     setIsLoggingIn(false);
 
+    // If no balance is loaded, redirect to setup screen
     if (!eoaBalanceEth) {
       goto(SetupScreen.SetupEoaFundingIncomplete);
-    } else if (!masterSafe?.address) {
+      return;
+    }
+
+    // If no service is created for the selected agent
+    if (!isServiceCreatedForAgent) {
+      window.console.log(
+        `No service for chain ${selectedServiceOrAgentChainId}`,
+      );
+      goto(SetupScreen.AgentSelection);
+      return;
+    }
+
+    // if master safe
+    if (masterSafe?.address) {
       goto(SetupScreen.SetupCreateSafe);
-    } else {
-      gotoPage(Pages.Main);
+      return;
     }
+
+    gotoPage(Pages.Main);
   }, [
+    isBalanceLoaded,
+    isWalletsFetched,
+    isServiceCreatedForAgent,
     canNavigate,
     eoaBalanceEth,
-    goto,
-    gotoPage,
-    isBalanceLoaded,
     masterSafe?.address,
     selectedServiceOrAgentChainId,
     wallets?.length,
+    selectedService?.name,
```

Comment:
unsure if we should have it here, and maybe should also check `isServicesFetched`
but can simply ask everyone to wait for some time before logging in, too complex to figure this out on the fly